CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc

```diff
@@ -168,6 +247,73 @@ SiPixelLorentzAnglePCLWorker::SiPixelLorentzAnglePCLWorker(const edm::ParameterS
   max_depth_ = 400.;
   min_drift_ = -1000.;  //-200.;(iConfig.getParameter<double>("residualMax"))
   max_drift_ = 1000.;   //400.;
+        
+bufsize = 64000;
+//    create tree structure
+//    Barrel pixel
+SiPixelLorentzAngleTree_ = new TTree("SiPixelLorentzAngleTree_","SiPixel LorentzAngle tree", bufsize);
```

Comment:
@wweiphy the indentation looks off. Please run:
```
scramv1 b code-check
scramv1b code-format
```
in order to comply with mandatory code quality recommendations.

Reply:
@mmusich There are errors when I run 

scramv1 b code-format

gmake: *** [config/SCRAM/GMake/Makefile.coderules:102: tmp/slc7_amd64_gcc900/code-checks-logs/changed-files-all.log] Error 129
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2

but I'm not sure how to access the log file.

Reply:
I am not sure, this command always worked for me.
Can you try to rebase to a more recent release?

Reply:
@mmusich I've switched into CMSSW_12_0_0_pre3 but still the same:

gmake: *** [config/SCRAM/GMake/Makefile.coderules:103: tmp/slc7_amd64_gcc900/code-checks-logs/changed-files-all.log] Error 129
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2

Which release are you using? I could try on that.

Reply:
I am using generally the last IB, but it should really not matter.  
- does the code compile without any flags?
- A part from the message are the files formatted correctly after applying the code format lint? You can check with `git diff`

Reply:
@mmusich I've run 'scram b' and it worked fine.

When I run 'scramv1 b code-check', it returns:
'''
>> Local Products Rules ..... started
>> Local Products Rules ..... done
Unknow target code-check
'''

'git diff' also works 

'''
diff --git a/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc b/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc
index 55daa1a914d..8445ed36139 100644
--- a/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc
+++ b/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc
@@ -4,9 +4,7 @@
 // Class:      SiPixelLorentzAnglePCLWorker
 //
 /**\class SiPixelLorentzAnglePCLWorker SiPixelLorentzAnglePCLWorker.cc CalibTracker/SiPixelLorentzAnglePCLWorker/plugins/SiPixelLorentzAnglePCLWorker.cc
-
  Description: [one line class summary]
-
  Implementation:
      [Notes on implementation]
 */
@@ -55,7 +53,8 @@
 
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "../interface/SiPixelLorentzAngleCalibrationStruct.h"
-
+#include <TTree.h>
+#include <TFile.h>
'''

Reply:
@mmusich I've run `scram b` and it worked fine

```
>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Creating project symlinks
>> Done python_symlink
>> Compiling python modules python
>> All python modules compiled
>> Plugins of all types refreshed.
>> Done generating edm plugin poisoned information
```

When I run `scramv1 b code-checks` it returns:
```
>> Local Products Rules ..... started
>> Local Products Rules ..... done
Unknow target code-check
```

Reply:
`git diff` also works

```
diff --git a/CalibTracker/SiPixelLorentzAngle/python/SiPixelLorentzAnglePCLWorker_cfi.py b/CalibTracker/SiPixelLorentzAngle/python/SiPixelLorentzAnglePCLWorker_cfi.py
index 04e139b5aba..51f3c425477 100644
--- a/CalibTracker/SiPixelLorentzAngle/python/SiPixelLorentzAnglePCLWorker_cfi.py
+++ b/CalibTracker/SiPixelLorentzAngle/python/SiPixelLorentzAnglePCLWorker_cfi.py
@@ -1,12 +1,13 @@
 import FWCore.ParameterSet.Config as cms
 
 from DQMServices.Core.DQMEDAnalyzer import DQMEDAnalyzer
-SiPixelLorentzAnglePCLWorker = DQMEDAnalyzer( 
+SiPixelLorentzAnglePCLWorker = DQMEDAnalyzer(
     "SiPixelLorentzAnglePCLWorker",
     folder = cms.string('AlCaReco/SiPixelLorentzAngle'),
-    src = cms.InputTag("TrackRefitter"),    
-    binsDepth  = cms.int32(50),
-    binsDrift =        cms.int32(200),
+    fileName = cms.string('testrun.root'),
+    src = cms.InputTag("TrackRefitter"),
+    binsDepth    = cms.int32(50),
+    binsDrift =    cms.int32(200),
     ptMin = cms.double(3),
     normChi2Max = cms.double(2),
     clustSizeYMin = cms.int32(4),
```

Reply:
It's:
`scramv1 b code-checks` not `code-check`.
Anyway the whole point of running it is that `scram b code-format` should automatically apply the mandatory code changes to comply with the quality standards. As the indentation in your file looks odd, I am expecting it to change after you run the code formatting. Please confirm that's the case and push the automatic changes.

Reply:
@mmusich This time running `scramv1 b code-checks` I got 

```
>> Local Products Rules ..... started
>> Local Products Rules ..... done
warning: Not a git repository. Use --no-index to compare two paths outside a working tree
usage: git diff --no-index [<options>] <path> <path>

Diff output format options
    -p, --patch           generate patch
    -s, --no-patch        suppress diff output
    -u                    generate patch
    -U, --unified[=<n>]   generate diffs with <n> lines context
    -W, --function-context
                          generate diffs with <n> lines context
    --raw                 generate the diff in raw format
    --patch-with-raw      synonym for '-p --raw'
    --patch-with-stat     synonym for '-p --stat'
    --numstat             machine friendly --stat
    --shortstat           output only the last line of --stat
    -X, --dirstat[=<param1,param2>...]
                          output the distribution of relative amount of changes for each sub-directory
    --cumulative          synonym for --dirstat=cumulative
    --dirstat-by-file[=<param1,param2>...]
                          synonym for --dirstat=files,param1,param2...
    --check               warn if changes introduce conflict markers or whitespace errors
    --summary             condensed summary such as creations, renames and mode changes
    --name-only           show only names of changed files
    --name-status         show only names and status of changed files
    --stat[=<width>[,<name-width>[,<count>]]]
                          generate diffstat
    --stat-width <width>  generate diffstat with a given width
    --stat-name-width <width>
                          generate diffstat with a given name width
    --stat-graph-width <width>
                          generate diffstat with a given graph width
    --stat-count <count>  generate diffstat with limited lines
    --compact-summary     generate compact summary in diffstat
    --binary              output a binary diff that can be applied
    --full-index          show full pre- and post-image object names on the "index" lines
    --color[=<when>]      show colored diff
    --ws-error-highlight <kind>
                          highlight whitespace errors in the 'context', 'old' or 'new' lines in the diff
    -z                    do not munge pathnames and use NULs as output field terminators in --raw or --numstat
    --abbrev[=<n>]        use <n> digits to display SHA-1s
    --src-prefix <prefix>
                          show the given source prefix instead of "a/"
    --dst-prefix <prefix>
                          show the given destination prefix instead of "b/"
    --line-prefix <prefix>
                          prepend an additional prefix to every line of output
    --no-prefix           do not show any source or destination prefix
    --inter-hunk-context <n>
                          show context between diff hunks up to the specified number of lines
    --output-indicator-new <char>
                          specify the character to indicate a new line instead of '+'
    --output-indicator-old <char>
                          specify the character to indicate an old line instead of '-'
    --output-indicator-context <char>
                          specify the character to indicate a context instead of ' '

Diff rename options
    -B, --break-rewrites[=<n>[/<m>]]
                          break complete rewrite changes into pairs of delete and create
    -M, --find-renames[=<n>]
                          detect renames
    -D, --irreversible-delete
                          omit the preimage for deletes
    -C, --find-copies[=<n>]
                          detect copies
    --find-copies-harder  use unmodified files as source to find copies
    --no-renames          disable rename detection
    --rename-empty        use empty blobs as rename source
    --follow              continue listing the history of a file beyond renames
    -l <n>                prevent rename/copy detection if the number of rename/copy targets exceeds given limit

Diff algorithm options
    --minimal             produce the smallest possible diff
    -w, --ignore-all-space
                          ignore whitespace when comparing lines
    -b, --ignore-space-change
                          ignore changes in amount of whitespace
    --ignore-space-at-eol
                          ignore changes in whitespace at EOL
    --ignore-cr-at-eol    ignore carrier-return at the end of line
    --ignore-blank-lines  ignore changes whose lines are all blank
    --indent-heuristic    heuristic to shift diff hunk boundaries for easy reading
    --patience            generate diff using the "patience diff" algorithm
    --histogram           generate diff using the "histogram diff" algorithm
    --diff-algorithm <algorithm>
                          choose a diff algorithm
    --anchored <text>     generate diff using the "anchored diff" algorithm
    --word-diff[=<mode>]  show word diff, using <mode> to delimit changed words
    --word-diff-regex <regex>
                          use <regex> to decide what a word is
    --color-words[=<regex>]
                          equivalent to --word-diff=color --word-diff-regex=<regex>
    --color-moved[=<mode>]
                          moved lines of code are colored differently
    --color-moved-ws <mode>
                          how white spaces are ignored in --color-moved

Other diff options
    --relative[=<prefix>]
                          when run from subdir, exclude changes outside and show relative paths
    -a, --text            treat all files as text
    -R                    swap two inputs, reverse the diff
    --exit-code           exit with 1 if there were differences, 0 otherwise
    --quiet               disable all output of the program
    --ext-diff            allow an external diff helper to be executed
    --textconv            run external text conversion filters when comparing binary files
    --ignore-submodules[=<when>]
                          ignore changes to submodules in the diff generation
    --submodule[=<format>]
                          specify how differences in submodules are shown
    --ita-invisible-in-index
                          hide 'git add -N' entries from the index
    --ita-visible-in-index
                          treat 'git add -N' entries as real in the index
    -S <string>           look for differences that change the number of occurrences of the specified string
    -G <regex>            look for differences that change the number of occurrences of the specified regex
    --pickaxe-all         show all changes in the changeset with -S or -G
    --pickaxe-regex       treat <string> in -S as extended POSIX regular expression
    -O <file>             control the order in which files appear in the output
    --find-object <object-id>
                          look for differences that change the number of occurrences of the specified object
    --diff-filter [(A|C|D|M|R|T|U|X|B)...[*]]
                          select files by diff type
    --output <file>       Output to a specific file

gmake: *** [config/SCRAM/GMake/Makefile.coderules:103: tmp/slc7_amd64_gcc900/code-checks-logs/changed-files-all.log] Error 129
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

Do you know what's wrong? `git diff` does work but there's a warning `Not a git repository. Use --no-index to compare two paths outside a working tree`

Reply:
From where are you running it?
It should be done in `$CMSSW_BASE/src` so that it sees the `.git` file

Reply:
Thanks! It works now.